### PR TITLE
Add further requirements note

### LINKED ISSUE #23: Collection note should indicate that further dependencies may be required.
Can't transfer the issue from https://github.com/ansible/ansible/issues/74195 so doing the copy-pasta:

I am a contributor to the ansible.azcollection project. In order for those modules to execute, many azure specific packages must be installed via pip.
At the top of all of the online documentation for modules, there is a "Note" banner on installing the collection, example
https://docs.ansible.com/ansible/latest/collections/azure/azcollection/azure_rm_storageaccount_module.html
That banner is added to every module's doc in all of the collections in that doc site.

The problem we're running into is that newer users read that banner and believe that it satisfies all dependencies by doing the 'ansible-galaxy collection install ${collection} ' . Could some additional text be added to indicate that there may be further dependencies, and that they should click on the collection link to see the collection specific docs?

I would attempt a PR myself, but I have no idea how the docs are composed or where the banner message is being pulled from. If this is not the correct repo to report the issue against, let me know and I'll resubmit there.

### PR DESCRIPTION
**Pull request by felixfontein**
Fixes #23.
- See https://ansible.fontein.de/collections/community/general/alternatives_module.html#ansible-collections-community-general-alternatives-module for a module with requirements, and https://ansible.fontein.de/collections/community/general/apk_module.html#ansible-collections-community-general-apk-module for one without.
- https://ansible.fontein.de/collections/community/general/bitwarden_lookup.html#ansible-collections-community-general-bitwarden-lookup is a plugin with requirements, https://ansible.fontein.de/collections/community/general/cartesian_lookup.html#ansible-collections-community-general-cartesian-lookup one without.

### CODE REVIEW

File: src/antsibull_docs/data/docsite/plugin.rst.j2 [hunk 1]
```diff
@@ -98,6 +98,10 @@
     To check whether it is installed, run :code:`ansible-galaxy collection list`.
 
     To install it, use: :code:`ansible-galaxy collection install @{collection}@`.
+{%   if doc['requirements'] %}
+    You need further requirements to be able to use this {% if plugin_type == 'module' %}module{% else %}@{ plugin_type }@ plugin{% endif %},
+    see :ref:`the Requirements <ansible_collections.@{plugin_name}@_@{plugin_type}@_requirements>` for details.
```
Comment (samccann) [inline src/antsibull_docs/data/docsite/plugin.rst.j2:103]:
```suggestion
    see :ref:`Requirements <ansible_collections.@{plugin_name}@_@{plugin_type}@_requirements>` for details.
```

**Review by samccann — COMMENTED** [review]
on little nit. Clever approach!!

**Comment by felixfontein** [comment]
(The links are updated with the latest version of this PR.)

**Review by samccann — APPROVED** [review]
Can't judge the code but the output LGTM!

**Comment by felixfontein** [comment]
@samccann thanks a lot for testing this!

**Pull request merged**